EX-16373 - PrebidJS Adapter (#1)
EXADS/Prebid.js: exads/doc/README.md

> # exadsBidAdapter

Reviewer: As discussed, all the `exads/*` should be documented in Confluence.

Author: Sure Pierre.

As mentioned yesterday by @Sofiabarreira, she has already planned to write the documentation to put into Confluence after QA testing.

I'll give her all information and support for doing that.

Reviewer: @giuseppe-exads @Sofiabarreira OK, just remember to remove this content from the PR before merging it. It could be temporarily placed somewhere else if needed!

Author: You are right @broucz. I'll put it in another place before the PR.

> # exadsBidAdapter

Reviewer: Our `master` branch is "1316 commits behind" Prebidjs `master`. We should update our master before the final review and QA of our implementation.

Author: Yes fully agree. Checking all stuff with the last version of PrebidJS is mandatory.

After finishing code review I'll ask to guys to align master to the last version and then I'll align the branch feature with master.

Author: Just asked to do that into the slack channel.

Author: master and feature branches updated with last original master branch

> Exads PrebidJS Adapter

Reviewer 2: ```suggestion
EXADS PrebidJS Adapter
```

Author: done

> #### In order to mantain the adapter locally:

Reviewer 2: As a user, I am having a hard time understanding this text.
Is there a reason why you are using the continuous verb (i.e., 'changing', 'updating') instead of the imperative (i.e., 'change', 'update')? Is it a convention for README files I am not aware of? It's making it hard to understand the message you want to relay. 
On another note, when you say 'into', do mean to just say 'in'? For example, "You can find them into `./exads/examples` ", or "Updating the unit tests, they are into `./Prebid.js/test/spec/modules/exadsBidAdapter_spec.js` ".

Author: > Is there a reason why you are using the continuous verb (i.e., 'changing', 'updating')...

Changed from gerund to Imperative

> On another note, when you say 'into', do mean to just say

I mean HTML examples. I've just added additional information to clarifly

Reviewer 2: > I mean HTML examples. I've just added additional information to clarify

Yes, it should be "in" instead of "into" then :)

Author: done

> * Change the adapter code that you can find into `./Prebid.js/modules/exadsBidAdapter.js`

Reviewer 2: ```suggestion
* Change the adapter code that you can find in `./Prebid.js/modules/exadsBidAdapter.js`
```

Author: replaced "into" with others prepositions in others places (to see my last commits)

> * Update the unit tests, they are into `./Prebid.js/test/spec/modules/exadsBidAdapter_spec.js`

Reviewer 2: ```suggestion
* Update the unit tests, they are in `./Prebid.js/test/spec/modules/exadsBidAdapter_spec.js`
```

Author: done

* Run tlint and unit tests (to see the specific paragraph)
* Do manual tests (to see the specific paragraph)
* Build the new version of the adapter and all modules needed: `gulp build --modules=consentManagement,exadsBidAdapter`
* After that, you can use the prebidJS, merged with our module. 

>     * You can find it into:  `./build/dist/prebid.js`

Reviewer 2: ```suggestion
    * You can find it in:  `./build/dist/prebid.js`
```

Author: done

* Updating our examples. You can find them into `./exads/examples`

#### Lint and Unit tests
* Note: lint checks the official prebidJS rules.
* Also, to do the pull request to official prebidJS team, it is mondatory 80% or more of covarage. 
* To check the coverage, type:
  * `gulp test-coverage` and then
  * `gulp view-coverage`

#### Manual tests
* Copying all examples to use into the publisher test web site: `./exads/doc/README.md`
* Copying the prebidJS library containing the new changes of the adapter `./build/dist/prebid.js`

> * For each example, changing `<script async src="js/prebid.js"></script>` based on the publisher web site configuration 

Reviewer 2: ```suggestion
* For each example, changing `<script async src="js/prebid.js"></script>` based on the publisher website configuration 
```

Author: done

 * Note: if you need to debug the snippet code, comment the previous instruction and uncomment the `<script src="js/prebid.js"></script>` 
* Changing all params based on test scenarios (to see: `./Prebid.js/modules/exadsBidAdapter.md`)

> * Navigating to the publisher web site and test it

Reviewer 2: ```suggestion
* Navigating to the publisher website and test it
```

Author: done
